### ISSUE TITLE
Add data owner info and experiment title metadata to ISIS ORSO file

### ISSUE THREAD

**Issue by rbauststfc** [mantidproject/mantid#36684]
The data owner names and affiliations, and experiment title are part of the mandatory information required by the ORSO Reflectometry reduced file format specification. At the moment, `SaveISISReflectometryORSO` is not able to include this information in the output file because not all of it is available or being populated from the raw Nexus file (see some explanation on https://github.com/mantidproject/mantid/issues/31073). The options seem to be as follows:

- Add these values to the raw Nexus file. This would need to done by Controls and they had some data policy queries about this. We investigated these and confirmed that there are no problems with compliance with the ISIS Data Policy (see details at end of this description). On the Mantid side we would need to decide where to load the information in to the Mantid workspace and make relevant changes to achieve this. This is the option that we have discussed most with our scientists.

- We can get some of the information from ICAT via the `CatalogSearch` algorithm. The experiment title can be retrieved this way. ICAT contains data owner names, so presumably this information should also be possible to retrieve somehow. Affiliation information would need to be added to ICAT and made retrievable via a Mantid algorithm. This approach would require users to login to ICAT before calling the save algorithm, although this should only be necessary once to create a session that could remain open for subsequent calls. Calls to `CatalogSearch` seem to be very quick, so performance is likely to be OK. If there were issues with ICAT or if connection dropped out at any point then we would not be able to retrieve the information - I don't know how frequently this is likely to happen, though.

- We could consider providing parameters for users to pass these values directly. This has the advantage that it's likely to be the quickest implementation compared to the other options. However, there is more risk of human error and obviously more user effort required with manual entry.

This will require further discussion to conclude the best way forward.

**Data Privacy:**

The relevant policies are:

- ISIS Data Policy: https://www.isis.stfc.ac.uk/Pages/Data-Policy.aspx
- ISIS Privacy Policy: https://www.isis.stfc.ac.uk/Pages/Privacy.aspx

In summary, the data owner names, affiliations (institutions) and experiment title are all information that we state in the privacy policy may be made public. This information should be omitted for ICRD (i.e. commercial experiments) related files, though (in theory this should be possible by filtering on RB number and would be applied to the original Nexus file, so it would be for the Controls Team to implement this).
